context/notebooks.sh

```diff
@@ -0,0 +1,29 @@
+#!/usr/bin/env bash
+
+# Clones repos with notebooks & compiles notebook test dependencies
+# Requires environment variables:
+#    RAPIDS_BRANCH
+#    CUDA_VER
+#    PYTHON_VER
+
+set -euo pipefail
+
+NOTEBOOK_REPOS=(cudf cuml cugraph cuxfilter cuspatial cusignal xgboost-conda)
+
+mkdir -p /notebooks /dependencies
+for REPO in "${NOTEBOOK_REPOS[@]}"; do
+    echo "Cloning $REPO..."
+    git clone -b "${RAPIDS_BRANCH}" --depth 1 --single-branch "https://github.com/rapidsai/$REPO" "$REPO"
+    cp -rL "$REPO"/notebooks /notebooks/"$REPO"
+    if [ -f "$REPO/dependencies.yaml" ] && yq -e '.files.test_notebooks' "$REPO/dependencies.yaml" > /dev/null; then
+        echo "Running dfg on $REPO"
+        rapids-dependency-file-generator \
+            --config "$REPO/dependencies.yaml" \
+            --file_key test_notebooks \
+            --matrix "cuda=${CUDA_VER%.*};arch=$(arch);py=${PYTHON_VER}" \
+            --output conda > "/dependencies/${REPO}_notebooks_tests_dependencies.yaml"; \
+    fi
+done
+
+pushd "/dependencies"
+conda-merge ./*.yaml > /test_notebooks_dependencies.yaml
```

Comment:
I'm a bit surprised that this script doesn't actually _install_ the dependencies. How are people expected to use the notebooks image? It looks like running this script is the last thing that happens in the Dockerfile.

Reply:
There are three stages to the build: dependencies (which runs this scripts), base, & notebooks. The notebooks stage depends on both base & dependencies. It grabs the resulting `test_notebooks_dependencies.yaml` from the dependencies stage and installs them.

Reply:
I see, that makes sense. Thanks for the explanation.